### DISCUSSION TITLE
Same NNN config for all users and servers

### DISCUSSION THREAD

**Discussion by FunctionDJ** [jarun/nnn#1714] [Q&A] [answered]
Hi, i use nnn a lot in my home network and i'm the only physical user. However, i use `sudo -i` a lot because i'm constantly browsing and editing system files too. I also do that a lot on my home server by running nnn on it (after ssh).

I'd like to know how i could keep the same config for all users and root as well as on my server. I know that it's common to keep a .dotfiles Git repository and use that to manually "sync" your configs (in nnn's case it's shell environment variables i guess). I've tried that before but found it very tedious. Are there other options?

Could i maybe use my local nnn to browse my server easily to that i don't even need nnn on the server or would that greatly reduce performance compared to running nnn on the machine directly? (E.g. big directories, spinning rust)

And can plugins like image preview even work when using ssh?
And what if i want to use icons where i have to compile nnn - do i need to repeat the steps on every system?

Maybe this topic is less about nnn and more about how to use Linux in general, i apologize 😅.

**Comment by 0xACE** [comment, marked as answer]
im using a ui im not used to so forgive me if this has already been solved.


but really you could just write you own `nnn` wrapper script... my `nnn` is just a `sh` script that sets up all of the necessary configuration required for my personal `nnn` experience...


yes sshfs will mount the files as a filesystem, so for the preview plugin it wont matter other than the limitations of `sshfs`
